wallet/chainntfns.go

```diff
@@ -889,6 +890,24 @@ func (w *Wallet) VoteOnOwnedTickets(ctx context.Context, winningTicketHashes []*
 				ticketVoteBits = tvb
 			}
 
+			// When not on mainnet, randomly disapprove blocks based
+			// on the disapprove percent.
+			dp := w.DisapprovePercent()
+			if dp > 0 {
+				if w.chainParams.Net == wire.MainNet {
+					log.Warnf("block disapprove percent set on mainnet")
+				} else {
+					rand.Seed(time.Now().UnixNano())
+					if int(dp) > rand.Intn(100) {
+						log.Infof("Disapproving block %v voted with ticket %v",
+							blockHash, ticketHash)
+						// Set the BlockValid bit to zero,
+						// disapproving the block.
+						ticketVoteBits.Bits = ticketVoteBits.Bits >> 1 << 1
```

Comment:
While this works for this case, it's a pretty non-standard way of going about it.

Usually to unset a bit, what you do is perform a bitwise `and` operation with a negated mask of the bit. So, you'd do something like:

```go
const blockIsValidBit = uint32(0x01)
ticketVoteBits.Bits &= ^blockIsValidBit
```

This also has the advantage of working for any bit position.

Reply:
Done.